modules/e2ee/E2EEncryption.js

```diff
@@ -218,15 +218,16 @@ export class E2EEncryption {
 
     /**
      * Advances the current key by using ratcheting.
-     * TODO: not yet implemented, we are just rotating the key at the moment,
-     * which is a heavier operation.
      *
      * @private
      */
     async _ratchetKeyImpl() {
         logger.debug('Ratchetting key');
 
-        return this._rotateKey();
+        this._e2eeCtx.ratchet(this.conference.myUserId());
+
+        // TODO: how do we tell the olm adapter which might need to send the current ratchet key
```

Comment:
We can make the ratchet function resolve a promise with the ratchetted key. Then feed it to the olm adapter, without broadcasting it.

Reply:
for now we can ratchet here and then tell olm? The alternative is that we do away with the ratchet message to the worker, call setKey with our ratcheted bytes. Less operations + codepaths

Reply:
we'll go for the "ratchet here, then use setKey" approach